Client/src/ArmoniK.DevelopmentKit.Client.Common/Submitter/BaseClientSubmitter.cs

```diff
@@ -366,56 +372,74 @@ public void WaitForTasksCompletion(IEnumerable<string> taskIds)
   /// <summary>
   ///   Get the result status of a list of results
   /// </summary>
-  /// <param name="resultIds">Collection of result ids</param>
+  /// <param name="taskIds">Collection of task ids from which to retrieve results</param>
   /// <param name="cancellationToken"></param>
   /// <returns>A ResultCollection sorted by Status Completed, Result in Error or missing</returns>
-  public ResultStatusCollection GetResultStatus(IEnumerable<string> resultIds,
+  public ResultStatusCollection GetResultStatus(IEnumerable<string> taskIds,
                                                 CancellationToken   cancellationToken = default)
   {
-    var remainingIds = resultIds.ToHashSet();
+    var mapTaskResults = TaskService.GetResultIds(new GetResultIdsRequest
```

Comment:
I would add a method of the baseclientsubmitter to do the conversion, and call it every time we need the result id. This method would do some caching.